File: rust/src/models/custom_property_definition_list_response.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{
    custom_property_definition::CustomPropertyDefinition, pagination_response::PaginationResponse,
};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct CustomPropertyDefinitionListResponse {
    pub data: Vec<CustomPropertyDefinition>,

    pub pagination_meta: PaginationResponse,
}

impl CustomPropertyDefinitionListResponse {
    pub fn new(data: Vec<CustomPropertyDefinition>, pagination_meta: PaginationResponse) -> Self {
        Self {
            data,
            pagination_meta,
        }
    }
}

File: typescript/src/models/customPropertyDefinitionListResponse.ts
// this file is @generated
import {
  type CustomPropertyDefinition,
  CustomPropertyDefinitionSerializer,
} from "./customPropertyDefinition";
import {
  type PaginationResponse,
  PaginationResponseSerializer,
} from "./paginationResponse";

export interface CustomPropertyDefinitionListResponse {
  data: CustomPropertyDefinition[];

  paginationMeta: PaginationResponse;
}

export const CustomPropertyDefinitionListResponseSerializer = {
  _fromJsonObject(object: any): CustomPropertyDefinitionListResponse {
    return {
      data: object["data"].map((item: any) =>
        CustomPropertyDefinitionSerializer._fromJsonObject(item)
      ),
      paginationMeta: PaginationResponseSerializer._fromJsonObject(
        object["pagination_meta"]
      ),
    };
  },

  _toJsonObject(self: CustomPropertyDefinitionListResponse): any {
    return {
      data: self.data.map((item: any) =>
        CustomPropertyDefinitionSerializer._toJsonObject(item)
      ),
      pagination_meta: PaginationResponseSerializer._toJsonObject(self.paginationMeta),
    };
  },
};
